.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,42 @@
+name: Run Gradle tests
+
+on: push
+
+env:
+  artifact_path: /tmp/external-system-test-api.zip
+  PANTS_SHA: "1.25.x-twtr"
+  IJ_ULTIMATE: false
+  CI: true
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v1
+      - name: Prepare
+        run: sudo apt-get install -y openjdk-8-jdk openjdk-11-jdk unzip curl python3-dev python3-virtualenv pkg-config git g++ libssl-dev make g++ zip
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'zulu'
+          java-version: '11'
+      - name: Cache .cache
+        uses: actions/cache@v2
+        with:
+          path: .cache
+          key: gradle-properties-${{ hashFiles('gradle.properties') }}
+      - name: Setup CI
+        run: PANTS_SHA=${{ env.PANTS_SHA }} ./scripts/setup-ci-environment.sh
+      - name: Test
+        run: |
+          export CWD=$(pwd)
+          export OSS_PANTS_HOME="$CWD/.cache/pants"
+          export DUMMY_REPO_HOME="$CWD/.cache/dummy_repo"
+          export JDK_LIBS_HOME="$CWD/.cache/jdk-libs"
+          ./gradlew :test
+      - name: Save build report if tests failed
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: build-reports
+          path: 'build/reports/'
+          retention-days: 2
```

Comment:
Personally, I'd prefer onger retention time and unconditional save

Reply:
Also wondering if it would be nice to upload plugin bundle as well

Reply:
Extended retention time, added unconditional save